lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java

```diff
@@ -170,9 +167,7 @@ public InputStream getInputStream() throws IOException {
 					File workingCopy = null;
 					try {
 						File file = new File(uri);
-						workingCopy = File.createTempFile("workingCopy", '.' + file.getName(), file.getParentFile());
-						Files.write(workingCopy.toPath(), document.getText().getBytes(), StandardOpenOption.CREATE);
-						Model model = mavenReader.read(new FileReader(workingCopy));
+						Model model = mavenReader.read(new ByteArrayInputStream(document.getText().getBytes()));
```

Comment:
This stream needs to be closed, or to be part of a try-with-resources block
```java
File file = new File(uri);
try (InputStream documentStream = new ByteArrayInputStream(document.getText().getBytes()) {
  Model model = mavenReader.read(documentStream);
  ...
}
```

Reply:
I thought about it is as well. But then again the Javadoc in ByteArrayInputStream doesn't hold any resources which need closing, it just backed by a byte array. Even the Javadoc says the close operation of ByteArrayInputStream does nothing. It's basically an empty method. So do we really need to complex the code here with the try with resource ?

Reply:
I'm afraid some static analysis engines or others can be confused because it's a Closeable and everyone/thing expects Closeable to be closed even if the Javadoc says it's not necessary for a given specific case.
The try-with-resources doesn't make code more complex actually, it extract the stream as a variable and wraps it in a try(...). IMO, it's better to make code seems more correct to automated or human readers, for the very small difference it makes in readability.

Reply:
agree I will fix it